src/engine/Server.cpp

```diff
@@ -113,6 +113,27 @@ Awaitable<void> Server::process(
     }
   }
 
+  // Set index name (which should be better called "description") and return
+  // stats (which contain the new name). TODO: Restrict this access by a token.
+  if (params.contains("index-description")) {
+    const auto& description = params.at("index-description");
+    LOG(INFO) << "Setting index description to: \"" << description << "\""
+              << std::endl;
+    _index.setKbName(description);
+    auto response = createJsonResponse(composeStatsJson(), request);
+    co_return co_await sendWithCors(std::move(response));
+  }
+
+  // Same for text index.
+  if (params.contains("text-description")) {
+    const auto& description = params.at("text-description");
```

Comment:
Several refactorings can be applied here (the duplication of the key is never good, and below the pattern of creating a JSON response and then sending it also occurs very often in this function.
Should we factor it out in this PR, or do you prefer a different one?